src/GSoft.Extensions.MediatR.Analyzers/Internals/RoslynExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace GSoft.Extensions.MediatR.Analyzers.Internals;
+
+internal static class RoslynExtensions
+{
+    public static INamedTypeSymbol? GetBestTypeByMetadataName(this Compilation compilation, string typeMetadataName, string? assemblyName = null)
+    {
+        var assemblies = GetAllAssemblies(compilation);
+
+        if (assemblyName != null)
+        {
+            assemblies = assemblies.Where(x => x.Name == assemblyName);
```

Comment:
Should we use `StringComparison` here for `string` equality or is this unnecessary?

Reply:
Using `stringA == stringB` is the equivalent of doing `string.Equals` with an ordinal case-sentive and culture-insensitive comparison.

Reference: https://learn.microsoft.com/en-us/dotnet/api/system.string.op_equality?view=netstandard-2.0#remarks

Reply:
Works for me 😁